elasticdl/python/tests/optimizer_wrapper_test.py

```diff
@@ -333,6 +473,67 @@ def test_report_to_kv_store(self):
                 expected_v, _ = expected_mock_kv_store.lookup([key])
                 self.assertTrue((v[0] == expected_v[0]).all())
 
+    def _random_init_model_weight(self, shapes, random_seed):
+        np.random.seed(random_seed)
+        return [np.random.rand(*shape).astype(np.float32) for shape in shapes]
+
+    def _test_correctness(self, optimizer_class, *args, **kwargs):
+        _model_file = get_module_file_path(
+            os.path.dirname(os.path.realpath(__file__)),
+            "embedding_test_module.KerasEmbeddingModel",
+        )
+        model_module = load_module(_model_file).__dict__
+
+        seed = 1
+        X, Y = _prepare_random_data(4, 4, 6, 4, True, random_seed=seed)
+        weights = self._random_init_model_weight(
+            [(4, 4), (4, 4), (72, 1), (1,)], seed
+        )
+        loss_fn = model_module["loss"]
+        model1 = model_module["KeraEmbeddingModel"](4, 4, weights)
+        opt1 = optimizer_class(*args, **kwargs)
+        _train(model1, opt1, X, Y, loss_fn, random_seed=seed)
+
+        model2 = model_module["EdlEmbeddingModel"](4, weights[2:])
+        opt2 = optimizer_class(*args, **kwargs)
+
+        layers = []
+        for layer in model2.layers:
+            if isinstance(layer, Embedding):
+                layers.append(layer.name)
+
+        embed_dims = dict([(layer_name, 4) for layer_name in layers])
+
+        # intialize embedding vectors in kv store
+        mock_kv_store = MockKvStore({})
+        for layer, embed_table in zip(layers, weights[:2]):
+            for i, embed_vector in enumerate(embed_table):
+                mock_kv_store.update(["%s-%d" % (layer, i)], [embed_vector])
+
+        with mock.patch.object(
+            EmbeddingService, "lookup_embedding", mock_kv_store.lookup
+        ), mock.patch.object(
+            EmbeddingService, "update_embedding", mock_kv_store.update
+        ):
+            _train_edl_embedding_with_optimizer_wrapper(
+                model2, opt2, X, Y, loss_fn, embed_dims, random_seed=seed
+            )
+
+        for layer1, layer2 in zip(model1.layers, model2.layers):
+            if "embedding" in layer2.name:
+                w1 = layer1.weights[0].numpy()
+                keys = [Embedding.get_key([layer2.name, i]) for i in range(4)]
+                w2 = np.concatenate(mock_kv_store.lookup(keys)[0]).reshape(
+                    4, -1
+                )
+                self.assertTrue((w1 - w2 < 0.0001).all())
+            else:
+                for w1, w2 in zip(layer1.weights, layer2.weights):
+                    self.assertTrue((w1 - w2 < 0.0001).numpy().all())
+
+    def test_correctness(self):
+        self._test_correctness(SGD, 0.1, 0.5)
```

Comment:
Use `kwargs` to pass arguments so it is clear what these arguments are.

Reply:
Fixed.